code/counter-app/src/main.rs

```diff
@@ -0,0 +1,175 @@
+// ANCHOR: imports
+use crossterm::event::{self, Event, KeyCode, KeyEvent, KeyEventKind};
+use ratatui::{prelude::*, widgets::*};
+// ANCHOR_END: imports
+
+// ANCHOR: modules
+mod tui;
+// ANCHOR_END: modules
+
+// ANCHOR: main
+fn main() -> color_eyre::Result<()> {
+    color_eyre::install()?;
+    tui::install_panic_hook();
+    let mut terminal = tui::init()?;
+    let app_result = App::default().run(&mut terminal);
+    tui::restore()?;
+    app_result
+}
+// ANCHOR_END: main
+
+// ANCHOR: app
+#[derive(Debug, Default)]
+pub struct App {
+    counter: u8,
```

Comment:
Why not use `usize`? Or even better `i64`? If a user presses and holds the key to increment or decrement, using `i64` would prevent runtime errors.

Reply:
I actually want the runtime error to happen to show color_eyre and the panic handler working.

Reply:
Ah that makes sense!